windows/src/engine/kmcomapi/com/system/keymancontrol.pas

```diff
@@ -690,6 +685,29 @@ procedure TKeymanControl.Do_Keyman_Exit;
 
 procedure TKeymanControl.LoadKeyman32;
 
+    function GetKeyman32Name: string;
+    var
+      Keyman32Name: string;
+    begin
+      Keyman32Name := '';
+      with TRegistryErrorControlled.Create do  // I2890
+      try
+        RootKey := HKEY_LOCAL_MACHINE;
+        if OpenKeyReadOnly(SRegKey_KeymanEngine_LM) and ValueExists(SRegValue_Keyman32_Name) then
+            Keyman32Name := ReadString(SRegValue_Keyman32_Name);
+      finally
+        Free;
+      end;
+
+      if Keyman32Name = '' then
+      begin
+        Keyman32Name := 'keyman32.dll';
```

Comment:
Is this safe?

Reply:
Thinking about it further we don't really want to use keyman32.dll for backwards compatibility because if this build of keyman doesn't find a registry entry something is wrong. 
Will just return an empty string if not found.